### PR TITLE
Setup Flipper

### LINKED ISSUE #6305: Setup Flipper
Just to track the work which is done on #6300

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

- Setup flipper on the project, for debug builds only (https://fbflipper.com/)
- Move getLastSession() to the SessionManager
- Create `DebugService` on the SDK to get access to the `Realmconfiguration`s
- Move `logDbUsageInfo()` to `DebugService`

What will be done later:
- write documentation (the file is here, but quite empty). This PR is created to unblock others dev faster.

## Motivation and context

Help developer to develop the app, especially by inspecting the databases content.

Closes #6305 

### Network

image

### Database

image

### LayoutInspector

image

### On phone Diagnostic Activity

image

## Tests

- Run the app in debug and use Flipper app
- Build the app in release mode, to ensure that it compiles.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: docs/flipper.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Flipper
+
+TODO Write doc
```
Comment (bmarty) [inline docs/flipper.md:3]: Will do it in a dedicated PR.

File: vector/build.gradle [hunk 2]
```diff
@@ -484,6 +496,7 @@ dependencies {
         exclude group: 'com.google.firebase'
         exclude group: 'com.google.android.gms'
         exclude group: 'com.android.installreferrer'
+        exclude group: 'com.facebook.fbjni', module: 'fbjni-java-only'
```
Comment (bmarty) [inline vector/build.gradle:499]: We may have a pb here on release build, will check.

Comment (bmarty) [reply vector/build.gradle]: Done

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/debug/DebugService.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+interface DebugService {
+    /**
+     * Get all the available Realm Configuration.
+     */
+    fun getAllRealmConfigurations(): List<RealmConfiguration>
+
+    /**
+     * Prints out info on DB size to logcat.
+     */
+    fun logDbUsageInfo()
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/debug/DebugService.kt:33]: Note: it should probably return a String rather than printing things on logcat, but this is out of scope of this PR.

**Comment by bmarty** [comment]
exodus is not happy.
https://etip.exodus-privacy.eu.org/trackers/90cd60c3-d0ac-42b6-91ae-9709dd6b13d7/

We should run exodus on the release version. Will see what we can do.

**Review by ganfra — COMMENTED** [review]
Two remarks/questions. Otherwise the plugin has been really helpful to debug timeline, so thats a good test :)

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt [hunk 4]
```diff
@@ -65,4 +66,8 @@ data class MatrixConfiguration(
          * Thread messages default enable/disabled value.
          */
         val threadMessagesEnabledDefault: Boolean = false,
+        /**
+         * List of network interceptors, they will be added when building an OkHttp client.
+         */
+        val networkInterceptors: List<Interceptor> = emptyList(),
```
[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt:72]: See my latest commit 5c84fdcd106c8f60cf3ec3d7422dd523b0a0c714, the interceptors provided by the configuration have to be added at the end, to ensure to see all the request header.
Providing on OkHttp client make sense though, maybe for another dedicated PR?

[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixConfiguration.kt:72]: See my latest commit 5c84fdcd106c8f60cf3ec3d7422dd523b0a0c714, the interceptors provided by the configuration have to be added at the end, to ensure to see all the request header.
Providing on OkHttp client make sense though, maybe for another dedicated PR?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/debug/DebugService.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import io.realm.RealmConfiguration
+
+/**
+ * Useful methods to access to some private data managed by the SDK.
+ */
+interface DebugService {
+    /**
+     * Get all the available Realm Configuration.
+     */
+    fun getAllRealmConfigurations(): List<RealmConfiguration>
```
[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/debug/DebugService.kt:28]: The Matrix SDK does not know what a Flipper plugin is, only the debug version of module `vector` knows it. Also there is only one Plugin to set up for all the RealmDatabase inspection.

**Comment by bmarty** [comment]
Rebased, exodus from GH action should be fine, exodus from Buildkite is still supposed to fail.

**Review by ouchadam — APPROVED** [review]
